.github/workflows/preview.yml

```diff
@@ -6,20 +6,24 @@ on:
   pull_request:
     branches: ['*demo*']
 
+defaults:
+  run:
+    working-directory: demos
+
 jobs:
   deploy-preview:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout 🛎️
+      - name: Checkout repository 🛎️
         uses: actions/checkout@v3
       - name: Install and build 🔧
-        run: npm run build:demos
+        run: npm run predeploy
       - name: Run tests ✅
-        run: npm run test:demo-react-ts
+        run: npm run test
       - name: Deploy preview 🚀
         uses: rossjrw/pr-preview-action@v1.3.0
         with:
-          source-dir: ./build
+          source-dir: demos/build
```

Comment:
> When more than one default setting is defined with the same name, GitHub uses the most specific default setting. For example, a default setting defined in a job will override a default setting that has the same name defined in a workflow.

This `source-dir` or `folder` config seems to override `defaults.run.working-directory` while testing due to the GH deploy action: https://github.com/JamesIves/github-pages-deploy-action/blob/43bb4e669776b52c5d8ace15b78b0a55b41864aa/action.yml#L39